jconstraints-core/build.gradle.kts

```diff
@@ -36,7 +36,7 @@ dependencies {
     implementation("commons-cli:commons-cli:1.4")
     implementation("dk.brics:automaton:1.12-1")
     implementation("org.antlr:antlr-runtime:3.5.2")
-    implementation("org.apache.commons:commons-math3:3.6.1")
+    api("org.apache.commons:commons-math3:3.6.1")
```

Comment:
Make sure, the fat jar of the jconstraints-runner artifact still bundles the comons-math3 jar and remains standalone executable as single jar. I have not read up the expected effects, but I don't see why this should be changed to api and commons-cli not, if this does not influence, how we package the jars. Just saying.

Reply:
That said, feel free to ignore my comment if you have verified that the jar remains runnable.

Reply:
Looks fine to me.

```
$ jar tf jconstraints-runner-all-0.9.6-SNAPSHOT.jar org/apache/commons/math3/
org/apache/commons/math3/
org/apache/commons/math3/ml/
org/apache/commons/math3/ml/neuralnet/
org/apache/commons/math3/ml/neuralnet/MapUtils.class
org/apache/commons/math3/ml/neuralnet/Network$NeuronIdentifierComparator.class
[…]
$ java -jar jconstraints-runner-all-0.9.6-SNAPSHOT.jar --smt /path/to/ARI5901.smt2 --solver z3
RESULT: UNSAT
$ java -jar jconstraints-runner-all-0.9.6-SNAPSHOT.jar --smt  /path/to/ARI5901.smt2 --solver cvc4
RESULT: UNSAT

```

The change to `api` is IMO necessary because core exposes `BigFraction` via `BuiltinTypes.RealType` to dependants. When scoping with `implementation`, the class is *not* available on the compiler classpath, i.e., `RealType` does not “work”. Consider, e.g.
```
import org.apache.commons.math3.fraction.BigFraction;
Constant.create(REAL, new BigFraction(1, 2)));
```
which should be valid usage, but does not compile. Each dependant that wants to use reals needs to `implementation`-depend on commons-math, i.e. we replicate the dependency in all subprojects (possible version conflicts, etc.).

Reply:
I get your point, good catch!